ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -154,6 +154,11 @@ func (r *RayServiceReconciler) Reconcile(ctx context.Context, request ctrl.Reque
 		If only one ray cluster exists, do serve deployment if needed and check dashboard, serve deployment health.
 		If both ray clusters exist, update active cluster status and do the pending cluster deployment and health check.
 	*/
+	if activeRayClusterInstance == nil && pendingRayClusterInstance == nil {
```

Comment:
1. If active cluster is nil, `Status.PendingServiceStatus.RayClusterName` will be set.
2. L141 - L149

These two logics promise that active / pending clusters will not be nil at the same time.